plugins/plugin-pullrequest/codenvy-plugin-pullrequest-ide/src/main/java/com/codenvy/plugin/pullrequest/client/ContributionMixinProvider.java

```diff
@@ -0,0 +1,135 @@
+                                     ContributePartPresenter contributionPartPresenter) {
+        this.vcsServiceProvider = vcsServiceProvider;
+        this.vcsHostingServiceProvider = vcsHostingServiceProvider;
+        this.workflowExecutor = workflowExecutor;
+        this.contributionPartPresenter = contributionPartPresenter;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void intercept(Resource resource) {
+        if (!resource.isProject()) {
+            return;
+        }
+
+        if (resource.getLocation().segmentCount() != 1) {
+            return;
+        }
+
+        if (vcsServiceProvider.getVcsService((Project)resource) == null) {
+            return;
+        }
+
+        final Project project = (Project)resource;
+
+        if (project.getMixins().contains(CONTRIBUTION_PROJECT_TYPE_ID)) {
+            return;
+        }
+
+        vcsHostingServiceProvider.getVcsHostingService(project)
+                                 .then(new Operation<VcsHostingService>() {
+                                     @Override
+                                     public void apply(final VcsHostingService vcsHostingService) throws OperationException {
+                                         addMixin(project)
+                                                 .then(new Operation<Project>() {
+                                                     @Override
+                                                     public void apply(Project project) throws OperationException {
+                                                         contributionPartPresenter.open();
+                                                         workflowExecutor.init(vcsHostingService, project);
+                                                     }
+                                                 })
+                                                 .catchError(new Operation<PromiseError>() {
+                                                     @Override
+                                                     public void apply(final PromiseError error) throws OperationException {
+                                                         workflowExecutor.invalidateContext(project);
+                                                     }
+                                                 });
+                                     }
+                                 })
+                                 .catchError(new Operation<PromiseError>() {
+                                     @Override
+                                     public void apply(final PromiseError error) throws OperationException {
+                                         workflowExecutor.invalidateContext(project);
+                                     }
+                                 });
+    }
+
+    private Promise<Project> addMixin(final Project project) {
+        if (!project.getMixins().contains(CONTRIBUTION_PROJECT_TYPE_ID)) {
```

Comment:
Can you please tell me what is the need for inverting the condition here, seems like we can avoid using negation?